src/interactivity/onecore/ConsoleInputThread.cpp

```diff
@@ -18,24 +18,23 @@ using namespace Microsoft::Console::Interactivity::OneCore;
 
 DWORD WINAPI ConsoleInputThreadProcOneCore(LPVOID /*lpParam*/)
 {
-    Globals& globals = ServiceLocator::LocateGlobals();
-    ConIoSrvComm* const Server = ConIoSrvComm::GetConIoSrvComm();
+    auto& globals = ServiceLocator::LocateGlobals();
+    const auto Server = ConIoSrvComm::GetConIoSrvComm();
 
-    NTSTATUS Status = Server->Connect();
+    auto Status = Server->Connect();
 
     if (NT_SUCCESS(Status))
     {
-        USHORT DisplayMode = Server->GetDisplayMode();
+        const auto DisplayMode = Server->GetDisplayMode();
 
         if (DisplayMode != CIS_DISPLAY_MODE_NONE)
         {
             // Create and set the console window.
-            ConsoleWindow* const wnd = new (std::nothrow) ConsoleWindow();
-            Status = NT_TESTNULL(wnd);
+            static ConsoleWindow wnd;

```

Comment:
`ConsoleWindow` also has no state and technically can be `constexpr`.
`SetConsoleWindowInstance` needs a mutable pointer however.